### PR TITLE
ENG-12354, inject random sleep into message handling loop to simulate network delay.

### PR DESCRIPTION
**Pull request by luyangco**
Change-Id: I1eac15480b301d892fb5b16f358e240f238f49e8

### CODE REVIEW

File: tests/frontend/org/voltcore/agreement/MiniSite.java [hunk 1]
```diff
@@ -50,16 +51,24 @@
...
         m_siteLog = logger;
         m_initialHSIds.addAll(HSIds);
         m_currentHSIds.addAll(HSIds);
         m_mailbox = mbox;
         m_arbiter = new MeshArbiter(mbox.getHSId(), mbox, this);
         m_failedHosts = callback;
+        long seed = System.currentTimeMillis() + mbox.getHSId();
+        if (givenSeed != 0L) {
+            seed = givenSeed;
+        }
```
Comment (zeemanhe) [inline tests/frontend/org/voltcore/agreement/MiniSite.java]: All the seeds in the tests are zero, is there a need to pass int the seed?  can just use System.currentTimeMillis() as seed to reduce all the changes?

File: tests/frontend/org/voltcore/agreement/TestFuzzMeshArbiter.java [hunk 2]
```diff
@@ -72,18 +72,20 @@ MiniNode getNode(int i)
...
             long HSId = getHSId(i);
             m_nodes.put(HSId, null);
+            seedMap.put(HSId, seeds[i]);
         }
         Set<Long> HSIds = m_nodes.keySet();
         for (long HSId : HSIds) {
-            m_nodes.put(HSId, new MiniNode(HSId, HSIds, m_fakeMesh));
+            m_nodes.put(HSId, new MiniNode(HSId, HSIds, m_fakeMesh, seedMap.get(HSId)));
             m_nodes.get(HSId).start();
         }
```
Comment (zeemanhe) [inline tests/frontend/org/voltcore/agreement/TestFuzzMeshArbiter.java]: This method should be simplified.  One loop should be enough. local variable seedMap is not needed.

File: tests/frontend/org/voltcore/agreement/MiniSite.java (hunk 1 above)
Comment (luyangco) [inline tests/frontend/org/voltcore/agreement/MiniSite.java]: Like the comments say, the seed array in these tests is a convenient way to reproduce any issue. If the unit test failed in Jenkins, you can grab the seeds printed in log files, fill in the array and re-run the test. It makes random test easier to reproduce.

File: tests/frontend/org/voltcore/agreement/TestFuzzMeshArbiter.java (hunk 2 above)
Comment (luyangco) [inline tests/frontend/org/voltcore/agreement/TestFuzzMeshArbiter.java]: No I can't. MiniNode needs to know all the peers, which the list can only be get after 1st for loop.

Comment (zeemanhe) [inline tests/frontend/org/voltcore/agreement/TestFuzzMeshArbiter.java]: I got it.  One way to simplify it a little bit is to get an array of HSIDs on the first loop.  Then loop through the array of HSIDs.  This way could avoid updating m_nodes twice and seedMap.

Comment (luyangco) [inline tests/frontend/org/voltcore/agreement/TestFuzzMeshArbiter.java]: OK. I'll make the change.

**Comment by luyangco** [comment]
Just use `System.currentTimeMillis()` is not enough, as all the MiniSites are started almost in the same time and every MiniSite needs to its own unique seed.

**Pull request closed without merging**